head/src/main/kotlin/io/qalipsis/core/head/report/thymeleaf/ThymeleafReportServiceImpl.kt

```diff
@@ -60,6 +61,8 @@ internal class ThymeleafReportServiceImpl(
     private val resourceLoader: ResourceLoader,
 ) : TemplateReportService {
 
+    private lateinit var cssFilePath: Path
```

Comment:
You had made the same mistake for the web reports. The service is a singleton, potentially used by 2 users at a time. What will happen when 2 people define a path concurrently here?

Reply:
This is a singleton, yes, and the `cssFilePath` is set immediately after the class is initialized and not by the user, given the `@PostConstruct`. Thus, this happens just once in its lifecycle.